Sources/NIOCore/CircularBuffer.swift

```diff
@@ -222,6 +222,33 @@ extension CircularBuffer: Collection, MutableCollection {
             return end.backingIndex &- start.backingIndex
         }
     }
+
+    @inlinable
+    public __consuming func _copyContents(
+        initializing buffer: UnsafeMutableBufferPointer<Element>
+    ) -> (Iterator,UnsafeMutableBufferPointer<Element>.Index) {
+        precondition(buffer.count >= self.count)
+
+        guard var ptr = buffer.baseAddress else {
+            return (self.makeIterator(), buffer.startIndex)
+        }
+
+        let indexRanges: [Range<Int>]
+        if self.tailBackingIndex >= self.headBackingIndex {
+            indexRanges = [self.headBackingIndex..<self.tailBackingIndex]
+        } else {
+            indexRanges = [ self.headBackingIndex..<self._buffer.count, 0..<self.tailBackingIndex ]
```

Comment:
We should prefer `self._buffer.endIndex`.

Reply:
Done.